### ISSUE TITLE
How to order API doc?

### ISSUE THREAD

**Issue by tarlepp** [nelmio/NelmioApiDocBundle#1120]
So basically order by `Swagger\Annotations\Tag` and route URL + method

**Comment by tarlepp** [comment]
Any solutions for this ?

**Comment by GuilhemN** [comment]
I'm not sure what you mean.

> order by Swagger\Annotations\Tag and route URL + method

Isn't this enough?

**Comment by tarlepp** [comment]
@GuilhemN so eg if I want to get something following:

GET /foo
GET /Foo/{id]
POST/foo
PUT /foo{id}
PATCH /foo{id}
DELETE /foo{id}

And currently I get something like
GET /user/{id}/roles
GET /user/{id}/groups
POST /user/{userId}/group/{userGroupId}
DELETE /user/{userId}/group/{userGroupId}
DELETE /user/{id}
GET /user/{id}
PATCH /user/{id}
PUT /user/{id}
GET /user/count
GET /user
POST /user
GET /user/ids

And sure you can see that there isn't structure point of that.

**Comment by GuilhemN** [comment]
I would have to check to be sure but I think this is ordered according to your routes definition order.
So I guess defining your routes in the order you prefer should do the job.

**Comment by GuilhemN** [comment]
Closing. I guess it's fixed now.

**Comment by AdrienBr** [comment]
Hello,
Same problem here, the operations are sorted in a very mysterious way. It doesn't match the Symfony route order. It's not an alphabetical order either.

It would be a great feature to configure explicitly the sorting of operations.

**Comment by p5yk0** [comment]
I guess it's ordered by tag then directory name then file name alphabetical order.

If you have this structure in your controllers directory (assuming they are all under the same tag) :
```
- one
  - ControllerZ.php
  - ControllerC.php 
- two
  - ControllerA.php
ControllerB.php 
```

Thus the endpoints will appear in this order :
- ControllerB.php 
- ControllerC.php
- ControllerZphp
- ControllerA.php

**Comment by webdevilopers** [comment]
Indeed the sorting could be optimized.

![swagger-ui-sort](https://user-images.githubusercontent.com/3479626/230723274-a66dc30d-830e-4a57-bb13-8755a687547d.png)

Our routes are directly defined on the controller classes. Even setting the tags and operation IDs correctly does not create the desired result.

![ide-controller-sort](https://user-images.githubusercontent.com/3479626/230723310-ceca0bef-ebbf-4e90-a12b-13d2db4028c6.png)

I think this will be fixed in #1330 by adding config parameters e.g. `operationsSorter` and `tagsSorter`:
https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md#display
